src/components/store/reducers/cart.ts

```diff
@@ -0,0 +1,91 @@
+import {createSlice, PayloadAction} from '@reduxjs/toolkit';
+import {Product} from '../../types/product';
+import {CartItemType} from "../../types/cartItemType";
+import {loadState} from "./storeDb";
+import {CartItemArgType} from "../../types/cartItemArgType";
+
+/**
+ * начальное состояние стейта
+ */
+let initialState = loadState<CartItemType[]>('cart', []);
+
+const slice = createSlice({
+  name: 'cart',
+  initialState,
+  reducers: {
+    /**
+     * добавляет продукт в стейт корзины, при повторном добавлении увеличивает quantity
+     */
+    addProductToCart: (state, { payload: cartItem }: PayloadAction<CartItemArgType>): CartItemType[] => {
+      const item = state.find((item) => item.product.id === cartItem.product.id);
```

Comment:
You use this line in several places, try to use common func for it
`const getItem = (state, pattern) => state.find((item) => item.product.id === pattern)`